Fix relative path handling for ragna panel app

aktech opened the pull request:
- [x] Fixes relative paths for panel apps

This doesn't handles the FastAPI paths yet, have just added an optional environment variable: `RAGNA_API_ROOT_PATH` for specifying root path for FastAPI server.

pmeier commented on ragna/deploy/_api/core.py:
```diff
@@ -37,7 +38,11 @@ def get_component(display_name: str) -> Component:
+    app = FastAPI(
+        title="ragna",
+        version=ragna.__version__,
+        root_path=os.environ.get("RAGNA_API_ROOT_PATH", ""),
```
IMO, this should be a config option. You just need to put

```python
root_path: str = ""
```

into

https://github.com/Quansight/ragna/blob/c08a223e5eb6450681cabd7cfd5843c30dbe8d60/ragna/deploy/_config.py#L55-L62

That way you can set it through the config file but also through the `RAGNA_API_ROOT_PATH` environment variable.

pmeier commented on ragna/deploy/_ui/js_utils.py:
```diff
@@ -0,0 +1,58 @@
+              console.log("Redirecting to " + redirectTo)
+              window.location.href = redirectTo;
+              if ({{remove_auth_cookie}}) {
+                console.log("Removing auth_token cookie")
```
Let's not.

```suggestion
```

This applies to all other `console.log`s as well.

pmeier commented on ragna/deploy/_ui/js_utils.py:
```diff
@@ -0,0 +1,58 @@
+    # so /foo/bar/car/auth becomes /foo/bar/car/
+    >>> redirect_script(remove="auth", append="/")
+    """
+    js_script = preformat(
```
IIUC, the only reason we need the `preformat` with six `str.replace` is because we want to use `str.format` later and our original text might already have curly braces inside of it, correct? If so, wouldn't it be easier to rely on `re.sub(r"\{\{\w+\}\}")` or whatever other delimiters for the substitution you want to use?

aktech replied:
So you mean without using `str.format`?

pmeier reviewed (commented):
Thanks Amit for the PR!

> This doesn't handles the FastAPI paths yet

I assume you mean "this doesn't handle FastAPI paths **automatically** yet", correct? I think that is ok. I don't see how the API would be able to detect this in the first place.

---

I have one question for the JS redirects: it seems we are doing mostly relative redirects, which have the disadvantage that the developer needs to know exactly at which path they currently are. Is there a technical blocker from always doing absolute redirects?

pmeier commented:
Does this PR resolve #275 or is there even more to the story?

aktech commented:
> I assume you mean "this doesn't handle FastAPI paths automatically yet", correct? I think that is ok. I don't see how the API would be able to detect this in the first place.

Yes, FastAPI would need a root api path, which I have added. If it needs more changes, then it's beyond the scope of this PR.

> Does this PR resolve https://github.com/Quansight/ragna/issues/275 or is there even more to the story?

Mostly, but I am not sure as I said I have not looked at the potential issues with FastAPI redirects. I want to keep the scope of this PR limited to panel redirects only. I'll create a separate PR for potential FastAPI changes if any. I added the root api path config option because it's anyway good to have and was an easy fix to potential redirect issues.

nenb reviewed (approved):
@pmeier I'm approving this as I don't see any major issues and any potential minor issues can likely be addressed in a subsequent PR.
